src/library/dao/directorydao.cpp

```diff
@@ -47,24 +65,7 @@ int DirectoryDAO::addDirectory(const QString& newDir) {
     return ALL_FINE;
 }
 
-bool DirectoryDAO::isChildDir(QString testDir, QString dirStr) {
-    QDir test = QDir(testDir);
-    QDir dir = QDir(dirStr);
-    bool child = dir == test;
-    while (test.cdUp()) {
-        if (dir == test) {
```

Comment:
I am afraid this loop is quite heavy because QT normalizes the Dir before compare, but does not cache the intermediate normalized values. I am not sure if disk IO is involved ...

I can think of a better appproach that normalizes the string first and than just find a sub string in a QString()

Reply:
We may also consider to combine these calls:   

```
    foreach (const QString& dir, dirs) {
        if (isChildDir(newDir, dir)) {
            childDir = dir;
        }
        if (isChildDir(dir, newDir)) {
            parentDir = dir;
        }
    }
```

On the other hand, you have just moved an existing function. Probably not worth the risk.

Reply:
I am not going to rewrite this legacy code, because I have already prepared a better solution. Still work in progress, not for 2.3.

Reply:
I'll leave this function as is, both the code **and** the name. It has only been moved from the class into an anonymous namespace.

Reply:
Before a rewrite we need tests!